common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -537,7 +540,9 @@ void closeAndDeleteOutdatedPartitions(
     partitions
       .forEach((partitionId, partitionInfo) -> {
         synchronized (partitionInfo) {
-          partitionInfo.closeAllFilesAndDeleteIfNeeded(true);
+          AppShufflePartitionInfo.closeAllFilesAndDeleteIfNeeded(true, partitionInfo.dataFile,
```

Comment:
```
CLEANER.register(this, new ResourceCleaner(dataFile, dataChannel, indexFile,
        metaFile, appAttemptShuffleMergeId, reduceId));
```
Maybe we can assign it to `Cleaner.Cleanable cleanable`, and then we can change this to call `partitionInfo.cleanable.clean();` there.

And then, IIRC,  if `partitionInfo.cleanable.clean();` is called manually,` ResouceCleaner.run` will not be executed repeatedly.